pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -217,7 +224,7 @@ public boolean readCompacted() {
     public Future<Void> sendMessages(final List<Entry> entries, EntryBatchSizes batchSizes,
```

Comment:
Add an overloading method here can avoid many changes

```java
public Future<Void> sendMessages(final List<Entry> entries, EntryBatchSizes batchSizes, EntryBatchIndexesAcks batchIndexesAcks,
                                     int totalMessages, long totalBytes, long totalChunkedMessages,
                                     RedeliveryTracker redeliveryTracker) {
    sendMessages(entries, batchSizes, batchIndexesAcks, totalMessages, totalBytes, totalChunkedMessages, redeliveryTracker, this.consumerEpoch);
}
```

Reply:
And the protocol handlers call this method, also can avoid modifications from the protocol handlers.